### DISCUSSION TITLE
URQL vs Apollo, CORS handling.

### DISCUSSION THREAD

**Discussion by jaymc-arg** [urql-graphql/urql#2739] [Q&A]
Hi everyone,
I'm trying to switch from Apollo to URQL, but I'm having a CORS problem when I try to connect to my backend wit URQL.

In Apollo i've done:
```
import {
  HttpLink,
  InMemoryCache,
  split,
  createHttpLink,
  ApolloLink,
} from "@apollo/client/core";
import { WebSocketLink } from "@apollo/client/link/ws";
import { getMainDefinition } from "@apollo/client/utilities";

import { GraphQLWsLink } from "@apollo/client/link/subscriptions";
import { createClient } from "graphql-ws";

const httpLink = createHttpLink({
  includeUnusedVariables: true, //incluir variables que no son reactivas por defecto via ()=>{}
  uri:
    process.env.API_GRAPHQL ||
    // Change to your graphql endpoint.
    "https://someSecureUrl.com/api/v1/graphql",
  headers: {
    Authorization:
      "Bearer someToken",
  },
});
...
```
This config continues with some cache setup, **and it works fine**.

While in URQL:
```
import urql, { createClient, provideClient } from "@urql/vue";
import { boot } from "quasar/wrappers";

const client = createClient({
  url: "https://someSecureUrl.com/api/v1/graphql",
  fetchOptions: () => {
    return {
      headers: {
        authorization:
          "Bearer  someToken",
      },
    };
  },
});

const urqlClient = provideClient(client);
export default boot(({ app }) => {
  app.use(urql, urqlClient);
  console.log("URQL", urqlClient);
});
```
I have no other configuration, and it's not any different from the options i'm using in Apollo, but I still receive:
![cors](https://user-images.githubusercontent.com/56963487/195710555-4cf1b09d-6f97-4e8d-a4b1-e0835309db20.png)

And a 504 status response.

Is there an additional configuration I have to do in URQL.

**Comment by JoviDeCroock** [comment]
Should be the same, only diff I see is the process.env and the difference in header capitalisation and your header value having an additional space.

Generally we can't change a whole lot about CORS errors as far as I can tell.

**Reply by jaymc-arg** [reply]
Nope, the process.env is the same, i've tried it. Same with Auth header.
Either way, i'm not even reaching the server due to CORS.
Tried it with spaceX public api and it works fine.
I don't have a clue...

**Comment by haouarihk** [comment]
from my experience 
- you need to re implement your caching system (starting from 0) 
- lazyQuery are just queries with pause option set to true,
- the respose from useQuery are flipped (apollo: [getdata, data], urql:[data, getdata])

still working on the authentication part, should be just an exchange, so i can't add more bullet points
